coreax/util.py

```diff
@@ -143,9 +144,7 @@ def apply_negative_precision_threshold(
     return jnp.where((-jnp.abs(precision_threshold) < _x) & (_x < 0.0), 0.0, _x)
 
 
-def pairwise(
-    fn: Callable[[ArrayLike, ArrayLike], Array],
-) -> Callable[[ArrayLike, ArrayLike], Array]:
+def pairwise(fn: Callable) -> Callable:
```

Comment:
This should definitely be given an improved type hint - adapt from the hint for `pairwise_fn` below?